File: src/dataset/cpp/simple-linked-list_exercism-toemm/claude/ClaudeSonnet4_simple_linked_list_v1.cpp
#include "simple_linked_list.h"

#include <stdexcept>

namespace simple_linked_list {

std::size_t List::size() const {
    return current_size;
}

void List::push(int entry) {
    Element* new_element = new Element{entry, this->head};
    this->head = new_element;
    ++this->current_size;
}

int List::pop() {
    if (current_size == 0) {
        throw std::domain_error("no elements");
    }
    
    Element* old_head = this->head;
    int data = old_head->data;
    this->head = old_head->next;
    delete old_head;
    --current_size;
    return data;
}

void List::reverse() {
    if (current_size <= 1) {
        return;
    }

    Element* prev = nullptr;
    Element* current = head;

    while (current != nullptr) {
        Element* next = current->next;
        current->next = prev;
        prev = current;
        current = next;
    }

    this->head = prev;
}

List::~List() {
    Element* current = this->head;
    while (current != nullptr) {
        Element* next = current->next;
        delete current;
        current = next;
    }
}

}  // namespace simple_linked_list

File: src/dataset/Java/simple-linked-list_exercism-ThomasZumsteg/claude/ClaudeSonnet4_SimpleLinkedList_v1.java
import java.util.NoSuchElementException;
import java.lang.reflect.Array;

public class SimpleLinkedList<T> {
    private Element<T> root = null;
    private int size = 0;

    //SimpleLinkedList creates a linked list with no values
    public SimpleLinkedList() { }

    public SimpleLinkedList(T[] values) {
        //SimpleLinkedList creates a linked list from an array of values
        for(int i = values.length - 1; i >= 0; i--) {
            Element<T> e = new Element<T>(values[i], root);
            root = e;
            size++;
        }
    }

   public Integer size() {
       //size the size of the linked list
       return size;
   } 

   public T pop() throws NoSuchElementException {
       //pop return and remove the top element of the linked list
       if(root == null)
           throw new NoSuchElementException();
       T value = root.value;
       root = root.next;
       size--;
       return value;
   }
       
   public void push(T value) {
       //push adds a new element to the top of the linked list
       root = new Element<T>(value, root);
       size++;
   }

   public T[] asArray(Class<T> type) {
       //asArray creates an array from the linked list
       T[] array = (T[])Array.newInstance(type, size);
       Element<T> e = root;
       for(int i = 0; e != null; e = e.next, i++)
           array[i] = e.value;
       return array;
   }

   public void reverse() {
       //reverse puts the elements of the linked list in the opposite order
       if(root == null || root.next == null) return;
       
       Element<T> prev = null;
       Element<T> current = root;
       Element<T> next;
       
       while(current != null) {
           next = current.next;
           current.next = prev;
           prev = current;
           current = next;
       }
       root = prev;
    }
}

class Element<T> {
    //Element is an element in the linked list
    public final T value;
    public Element<T> next;
    
    public Element(T value, Element<T> next) {
        this.value = value;
        this.next = next;
    }
}